### PR TITLE
Add periodicity to UNet using `CircularPad2d`

### LINKED ISSUE #56: 'Seaming' or fold in UNet outputs after passing smooth inputs
@TomMelt discovered discontinuities in the outputs as we do not enforce periodicity in the Unet. 

Image

### PR DESCRIPTION
**Pull request by ma595**
This PR aims to address the issue observed by @TomMelt, where a vertical 'seam' or fold is observed at the meridian line (lon = 0). We hadn't noticed this before as we were plotting data (lons 0-360). 

This is illustrated here in the raw flux outputs of the network. Tom has already verified that the inputs are smooth. 
image

This issue appears because the Unet does not enforce longitudinal periodicity. We can fix this by using the `CircularPad2d` function where convolution is done, i.e. in classes `Conv_block`, `Upsample` and `Attention_block`. 

I also recommend a replication pad at the poles using [nn.ReplicationPad2d](https://docs.pytorch.org/docs/stable/generated/torch.nn.ReplicationPad2d.html). This will eliminate potential steep gradients there. 

```
self.pad_layer = nn.Sequential(
            nn.CircularPad2d((padding, padding, 0, 0)),
            nn.ReplicationPad2d((0, 0, padding, padding))
        )
```

TODO:
- [x] Zero pad or replication pad in latitude? 
- [ ] Light testcase to ensure training will work. 
- [ ] Add some docs to code for clarity

Closes #56

### CODE REVIEW

File: utils/model_definition.py [hunk 1]
```diff
@@ -138,23 +138,31 @@ def totalsize(self):
...
+
         self.conv = nn.Sequential(
+            pad_layer,
             nn.Conv2d(
                 in_channels=ch_in,
                 out_channels=ch_out,
                 kernel_size=kernel_size,
                 stride=stride,
-                padding=padding,
+                padding=0,
```
Comment (TomMelt) [inline utils/model_definition.py]: just a reminder, can you remove these lines completely. The default for `Conv2d` is `padding=0`. https://docs.pytorch.org/docs/stable/generated/torch.nn.Conv2d.html
